src/Database/DBA.php

```diff
@@ -89,9 +91,15 @@ public static function connect($serveraddr, $user, $pass, $db, $charset = null)
 			}
 		}
 
-		if (!self::$connected && class_exists('mysqli')) {
+		if (!self::$connected && class_exists('\mysqli')) {
```

Comment:
Shouldn't this be better rewritten to check if the providing extension is loaded rather than the "bare" class exists? IMO this needs further rewriting, towards backend "drivers" (PDO, mysqli, etc. driver) instead of having hard-coded `switch()` or `if/elseif` blocks.

Reply:
But surely this rewrite leaves the scope of this PR. So this should be considered for an upcoming PR and not this one.

Reply:
OT: This class can then act as a "frontend" (or facade) class encapsulating away the database driver behind the scene.

Reply:
On paper, you're right. In practice, we only support MySQL-compatible databases, so we aren't anywhere near supporting an additional driver. We can revisit this organization when we actually add support for PostgreSQL for example.